### PR TITLE
Add trust_domain_id label to started metric

### PR DESCRIPTION
**Pull request by StupidScience**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [ ] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**

This PR adds `trust_domain_id` label to metric `started`

**Description of change**

This PR adds `trust_domain_id` label to metric `started`

**Which issue this PR fixes**

fixes https://github.com/spiffe/spire/issues/4120.

### CODE REVIEW

**Comment by StupidScience** [comment]
Failed test isn't connected to changes anyhow, prob some flakiness.

File: doc/telemetry.md [hunk 1]
```diff
@@ -6,79 +6,79 @@ The following metrics are emitted:
...
+| Counter      | `workload_api`, `connection`               |                            | The Workload API has successfully established a new connection.                       |
+| Gauge        | `workload_api`, `connections`              |                            | The number of active connections that the Workload API has.                           |
+| Sample       | `workload_api`, `discovered_selectors`     |                            | The number of selectors discovered during a workload attestation process.             |
+| Call Counter | `workload_api`, `workload_attestation`     |                            | The Workload API is performing a workload attestation.                                |
+| Call Counter | `workload_api`, `workload_attestor`        | `attestor`                 | The Workload API is invoking a given attestor.                                        |
+| Gauge        | `started`                                  | `version``trust_domain_id` | Information about the Agent.                                                          |
```
Comment (amartinezfayo) [inline doc/telemetry.md:80]:
```suggestion
| Gauge        | `started`                                  | `version`, `trust_domain_id` | Information about the Agent.                                                          |
```

**Review by amartinezfayo — COMMENTED** [review]
Thank you @StupidScience for this contribution!
After fixing the .md we should be ready to go.

File: doc/telemetry.md (hunk 1 above)
Comment (StupidScience) [reply doc/telemetry.md]: fixed, thanks

**Review by amartinezfayo — APPROVED** [review]
Thank you @StupidScience!

**Pull request merged**